MREUnityRuntime/MREUnityRuntimeLib/Core/Physics/JitterBuffer.cs

```diff
@@ -553,13 +591,28 @@ public void Update(ref RigidBodyData entry)
 					entry.MotionType = rb.MotionType;
 					entry.Transform = rb.Transform;
 					entry.Time = _snapshot.Time;
-
+					entry.SetVelocitiesToZero();
+					if (_prevIndex < _prevSnapshot.Transforms.Count &&
+						_prevSnapshot.Transforms[_prevIndex].RigidBodyId == entry.RigidBodyId)
+					{
+						// estimate velocity
+						float DT = _snapshot.Time - _prevSnapshot.Time;
+						if (Math.Abs(DT) > 0.0005F)
```

Comment:
>Math.Abs(DT) > 0.0005F [](start = 10, length = 22)

do we need this check?

Reply:
Yes, but we should do that within the function. Just to be more defensive if we end up somehow with the same two snapshots.

---
In reply to: [453666446](https://github.com/microsoft/mixed-reality-extension-unity/pull/218#discussion_r453666446) [](ancestors = 453666446)